dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1548,3 +1570,140 @@ async fn cmd_nexus_sled_expunge(
     );
     Ok(())
 }
+
+/// Runs `omdb nexus sleds expunge-disk`
+async fn cmd_nexus_sled_expunge_disk(
+    client: &nexus_client::Client,
+    args: &DiskExpungeArgs,
+    omdb: &Omdb,
+    log: &slog::Logger,
+    _destruction_token: DestructiveOperationToken,
+) -> Result<(), anyhow::Error> {
+    use nexus_db_queries::context::OpContext;
+
+    let datastore = args.db_url_opts.connect(omdb, log).await?;
+    let opctx = OpContext::for_tests(log.clone(), datastore.clone());
+    let opctx = &opctx;
+
+    // First, we need to look up the disk so we can lookup identity information.
+    let (_authz_physical_disk, physical_disk) =
+        LookupPath::new(opctx, &datastore)
+            .physical_disk(args.physical_disk_id.into_untyped_uuid())
+            .fetch()
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to find physical disk {}",
+                    args.physical_disk_id
+                )
+            })?;
+
+    // Helper to get confirmation messages from the user.
+    let mut line_editor = Reedline::create();
+    let mut read_with_prompt = move |message: &str| {
```

Comment:
What do you think about factoring some of this code out so it can be shared with `cmd_nexus_sled_expunge` above? Not sure how valuable that is but it looks relatively similar at first glance.